### PR TITLE
Upgrade to MSAL

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/jenkinsci/azure-ad-plugin/issues/60
Fixes https://github.com/jenkinsci/azure-ad-plugin/issues/55
Fixes https://github.com/jenkinsci/azure-ad-plugin/issues/75
Fixes https://github.com/jenkinsci/azure-ad-plugin/issues/53
Fixes https://github.com/jenkinsci/azure-ad-plugin/issues/88

This means no more guava conflicts yay as we aren't using the azure libraries for java SDK.

Graph SDK is 20MB though 😢, we could maybe use proguard on it to trim it back in the future if required 

TODO:

- [x] Review docs and ensure correct permissions are there
- [x] ~(possibly) improve searching if my stackoverflow questions get answered, users one works via API but not SDK, and groups searching doesn't work via either.~ Needed to add the consistencylevel header, bug in the graph explorer doesn't add it for code generation

@xuzhang3 you got time to look this one over?

Proxy support and sovereign cloud is untested, any feedback on that would be appreciated

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -76,6 +76,57 @@
...
+                    <groupId>com.fasterxml.jackson.dataformat</groupId>
+                    <artifactId>jackson-dataformat-xml</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <!-- should be kept in sync with jackson version in bom, possibly add it to the jackson2-api plugin -->
+        <dependency>
+            <groupId>com.fasterxml.jackson.dataformat</groupId>
+            <artifactId>jackson-dataformat-xml</artifactId>
```
Comment (timja) [inline pom.xml:127]: @jvz should this be added to the jackson2-api plugin?

Comment (jvz) [reply pom.xml:127]: Are any other plugins using that, too?

Comment (timja) [reply pom.xml:127]: not sure

Comment (jvz) [reply pom.xml:127]: Well, there's not much point in adding it to the common plugin if it's only used here.

**Review by xuzhang3 — APPROVED** [review]

**Comment by hikkyXII** [comment]
That might be only my issue, but Jenkins fails to log me in after this update(or one of 2/3 previous). 
With exception:
jenkins java.lang.NoClassDefFoundError: Could not initialize class com.azure.identity.implementation.IdentityClient
`
Checking another versions.

**Comment by timja** [comment]
Possibly a dependency conflict, this works in isolation.

What other azure plugins do you have installed?

**Comment by timja** [comment]
best to create an issue linking to this

**Pull request merged**